src/main/java/io/vertx/core/spi/cluster/ClusterManager.java

```diff
@@ -83,20 +75,56 @@
    */
   void nodeListener(NodeListener listener);
 
+  /**
+   * Store the details about this clustered node.
+   */
+  Future<Void> setNodeInfo(NodeInfo nodeInfo);
+
+  /**
+   * Get details about this clustered node.
+   */
+  NodeInfo getNodeInfo();
+
+  /**
+   * Get details about a clustered Vert.x node.
+   *
+   * @param nodeId the clustered node id
+   */
+  Future<NodeInfo> getNodeInfo(String nodeId);
```

Comment:
instead pass `Promise<NodeInfo>` so the `ClusterManager` does not need to care about which context to use

Reply:
done